### DISCUSSION TITLE
Cannot use Access Control name in Flow Permissions; must use UUID

### DISCUSSION THREAD

**Discussion by jonathan-chin** [directus/directus#24939] [Feature Requests]
### Describe the Improvement

I'm creating a bunch of flows and want to use some tight security practices. I have some `Read Data` flows that need only need to fetch information. I can set the Permissions to one of:

- From Trigger
- Public Role
- Full Access
- Other

None of the 3 predefined options really fit what I want. Instead, I've created a new `Access Control` level that only has read permissions on some collections. I've named it `Internal Read Only`. However, when trying to use that name in the `Other` option of Permissions, I get the following error:

code: 22P02
routine: string_to_uuid

When I replace the string inside `Other` with the UUID of this Role (eg: `fd364db4-bcc2-48f9-b0fe-f1b1213d7495`), it works. It would be a huge quality of life improvement if I can type in the Role's name; it would be even better if custom Roles populated here too.

**Comment by shipsides-sparknel** [comment]
love the idea of a M2O for that field if it was possible, but referring to a Role by it's GUID (which won't change) is safer than it's name (which could)

**Comment by jonathan-chin** [comment]
> love the idea of a M2O for that field if it was possible, but referring to a Role by it's GUID (which won't change) is safer than it's name (which could)

could it be designed so that the Role's name is populated into a dropdown but it saves the GUID?

when building a flow and adding a `Trigger Flow` node, the Flow field is a dropdown that's populated the same way. can Permissions be refactored the same way?

**Comment by jonathan-chin** [comment]
some other ideas for improvements:

1. maybe add another hardcoded option that takes the permissions from the previous node in the flow? currently, I'm reading data from 4 different collections, using 4 nodes, and they all have the same permissions.
2. there's the hardcoded option to take permissions `From Trigger`. however, I'm not sure how to control that? there's an initial node that allows the flow to be triggered from another flow. could I set permissions there? effectively, it would allow me to set the permissions for the whole flow.
